### PR TITLE
feat(lesson01): updates slides and Terraform format

### PR DESCRIPTION
**Pull request by dancortes94**
feat(lesson01): updates slides and Terraform format

### CODE REVIEW

File: lesson01/variables.tf [hunk 1]
```diff
@@ -23,9 +23,15 @@ variable "app_version" {
...
+variable "instance_key" {
+  default     = "academy-ohio"
+  description = "AWS key created to access the instance via ssh"
+  type        = "string"
+}
+
+# Only uncomment if you have a hosted zone in Route53
 # variable "domain" {
-#   default     = "mydomain.com"
+#   default     = "academy.wizeline.dev"
```
Comment (chitopunk) [inline lesson01/variables.tf]:
```academy.wizeline.dev.   21599   IN      NS      ns-1394.awsdns-46.org.
academy.wizeline.dev.   21599   IN      NS      ns-1549.awsdns-01.co.uk.
academy.wizeline.dev.   21599   IN      NS      ns-31.awsdns-03.com.
academy.wizeline.dev.   21599   IN      NS      ns-679.awsdns-20.net.```

File: slides/lesson1/solution/main.tf [hunk 2]
```diff
@@ -64,11 +57,11 @@ resource "aws_instance" "web" {
...
-  zone_id = data.aws_route53_zone.current.zone_id
-  name    = "${data.aws_caller_identity.current.user_id}.${var.domain}"
-  type    = "A"
-  ttl     = 300
-  records = [aws_instance.web.public_ip]
-}
-
+# Uncomment this if you have a hosted zone in your AWS account
+# resource "aws_route53_record" "dns_web" {
+#   zone_id = "${data.aws_route53_zone.current.zone_id}"
```
Comment (chitopunk) [inline slides/lesson1/solution/main.tf]: can you use terraform 0.12 format? regardless if this is not used

File: slides/lesson1/solution/main_hard_code.tf [hunk 3]
```diff
@@ -70,11 +64,11 @@ resource "aws_instance" "web" {
...
-  zone_id = data.aws_route53_zone.current.zone_id
-  name    = "${data.aws_caller_identity.current.user_id}.academy.wizeline.dev"
-  type    = "A"
-  ttl     = 300
-  records = [aws_instance.web.public_ip]
-}
-
+# Uncomment this if you have a hosted zone in your AWS account
+# resource "aws_route53_record" "dns_web" {
+#   zone_id = "${data.aws_route53_zone.current.zone_id}"
```
Comment (chitopunk) [inline slides/lesson1/solution/main_hard_code.tf]: same here, terraform 0.12 format

**Review by chitopunk — CHANGES_REQUESTED** [review]

File: lesson01/data.tf [hunk 4]
```diff
@@ -29,12 +29,10 @@ data "template_file" "user_data" {
...
-data "aws_caller_identity" "current" {
-}
+data "aws_caller_identity" "current" {}
 
-# Only uncommend if you have a hosted zone in Route53
+# Uncomment if you have a hosted zone on your AWS account
 # Gets hosted zone ID
 # data "aws_route53_zone" "current" {
-#   name = var.domain
+#   name = "${var.domain}"
```
Comment (estebancrw) [inline lesson01/data.tf]: Let's use Terraform v0.12 notation - `#   name = var.domain`

File: lesson01/variables.tf [hunk 5]
```diff
@@ -23,9 +23,15 @@ variable "app_version" {
...
 #   description = "The domain name to use"
-#   type        = string
+#   type        = "string"
```
Comment (estebancrw) [inline lesson01/variables.tf]: Let's use Terraform v0.12 notation - `#   type        = string`

File: lesson01/variables.tf [hunk 6]
```diff
@@ -23,9 +23,15 @@ variable "app_version" {
   type        = string
 }
 
-# Only uncommend if you have a hosted zone in Route53
+variable "instance_key" {
+  default     = "academy-ohio"
+  description = "AWS key created to access the instance via ssh"
+  type        = "string"
```
Comment (estebancrw) [inline lesson01/variables.tf]: Let's use Terraform v0.12 notation - `type        = string`

File: slides/lesson1/solution/main.tf [hunk 7]
```diff
@@ -64,11 +57,11 @@ resource "aws_instance" "web" {
...
+#   name    = "${data.aws_caller_identity.current.user_id}.${var.domain}"
+#   type    = "A"
+#   ttl     = 300
+#   records = ["${aws_instance.web.public_ip}"]
```
Comment (estebancrw) [inline slides/lesson1/solution/main.tf]: Let's use Terraform v0.12 notation - `#   records = [aws_instance.web.public_ip]`

File: slides/lesson1/solution/versions.tf [hunk 8]
```diff
@@ -1,4 +1,4 @@
 
 terraform {
   required_version = ">= 0.12"
-}
+}
```
Comment (estebancrw) [inline slides/lesson1/solution/versions.tf:4]: styling: let's add the blank line at the end of the file. This should be added automatically if you use [editorconfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) for VS Code

**Review by estebancrw — CHANGES_REQUESTED** [review]
Great work! Only styling changes

**Review by chitopunk — APPROVED** [review]

**Review by estebancrw — APPROVED** [review]
LGTM

**Pull request merged**